### ISSUE TITLE
Errors when using Fn::Transform

### ISSUE THREAD

**Issue by phene** [aws-cloudformation/cfn-python-lint#919]
*cfn-lint version: (`0.20.2`)*

Given the following valid template, I see 2 errors:

```yaml
Parameters:
  ImageId:
    Type: AWS::EC2::Image::Id
  InstanceType:
    Type: String

Resources:
  LaunchConfiguration:
    Type: AWS::AutoScaling::LaunchConfiguration
    Properties:
      ImageId: !Ref ImageId
      InstanceType: !Ref InstanceType
      UserData:
        'Fn::Base64':
          'Fn::Sub':
            'Fn::Transform': # Returns a string that contains Fn::Sub tokens like ${AWS::Region}
              Name: DynamicUserData
```
* `E1019 Sub should be a string or array of 2 items for Resources/LaunchConfiguration/Properties/UserData/Fn::Base64/Fn::Sub` - `Fn::Transform` can return a string or a template, so `Fn::Sub` should be forgiving of it.
* `E0002 Unknown exception while processing rule W2001: expected string or bytes-like object` Same root cause, but fails in a different way due to assumption that it complies with E1019 in a specific way.

**Comment by kddejong** [comment]
We will probably fail on higher level Transforms for a while as we need to be able to lint the transformed template.  However, in this case we shouldn't fail at all as this Transform is way down in the values of a properties section.
